src/Sarif.Converters/ConverterFactory.cs

```diff
@@ -0,0 +1,132 @@
+            if (!File.Exists(pluginAssemblyPath))
+            {
+                string message = string.Format(
+                    CultureInfo.CurrentCulture,
+                    ConverterResources.ErrorMissingPluginAssembly,
+                    pluginAssemblyPath);
+
+                throw new ArgumentException(message, nameof(pluginAssemblyPath));
+            }
+
+            string converterTypeName = toolFormat + "Converter";
+
+            Assembly pluginAssembly = Assembly.LoadFile(pluginAssemblyPath);
+            Type[] pluginTypes = pluginAssembly
+                .GetTypes()
+                .Where(t => t.IsPublic && t.Name.Equals(converterTypeName, StringComparison.Ordinal))
+                .ToArray();
+
+            if (pluginTypes.Length == 0)
+            {
+                string message = string.Format(
+                    CultureInfo.CurrentCulture,
+                    ConverterResources.ErrorMissingConverterType,
+                    pluginAssemblyPath,
+                    converterTypeName,
+                    toolFormat);
+
+                throw new ArgumentException(message, nameof(pluginAssemblyPath));
+            }
+
+            if (pluginTypes.Length > 1)
+            {
+                string message = string.Format(
+                    CultureInfo.CurrentCulture,
+                    ConverterResources.ErrorAmbiguousConverterType,
+                    pluginAssemblyPath,
+                    converterTypeName,
+                    toolFormat);
+
+                throw new ArgumentException(message, nameof(pluginAssemblyPath));
+            }
+
+            Type converterType = pluginTypes[0];
+            if (converterType.GetConstructor(
+                BindingFlags.DeclaredOnly | BindingFlags.Instance | BindingFlags.Public,
+                binder: null,
+                types: new Type[0],
+                modifiers: new ParameterModifier[0]) == null)
+            {
+                string message = string.Format(
+                    CultureInfo.CurrentCulture,
+                    ConverterResources.ErrorConverterTypeHasNoDefaultConstructor,
+                    converterType.FullName,
+                    pluginAssemblyPath,
+                    toolFormat);
+
+                throw new ArgumentException(message, nameof(pluginAssemblyPath));
+            }
+
+            var converter = Activator.CreateInstance(converterType) as ToolFileConverterBase;
+
+            if (converter == null)
+            {
+                string message = string.Format(
+                    CultureInfo.CurrentCulture,
+                    ConverterResources.ErrorIncorrectConverterTypeDerivation,
+                    converterType.FullName,
+                    pluginAssemblyPath,
+                    toolFormat,
+                    typeof(ToolFileConverterBase).FullName);
+
+                throw new ArgumentException(message, nameof(pluginAssemblyPath));
+            }
+
+            return converter;
+        }
+
```

Comment:
Everything from here on down is moved from ToolFormatConverter.cs, with some renamings.